core/templates/pages/topics-and-skills-dashboard-page/navbar/topics-and-skills-dashboard-navbar.directive.ts

```diff
@@ -78,8 +76,10 @@ angular.module('oppia').directive('topicsAndSkillsDashboardNavbar', [
                   var newExplanationObject = null;
 
                   $scope.$watch('newSkillDescription', function() {
-                    $scope.rubrics[1].setExplanation(
-                      '<p>' + $scope.newSkillDescription + '</p>');
+                    var rubrics = $scope.rubrics[1].getExplanations();
+                    rubrics[0] = '<p>' + $scope.newSkillDescription + '</p>';
```

Comment:
Ah, three things:

(a) Let's avoid constructing HTML in the JS. We really don't want to do that. Have all the HTML in Angular templates instead.

(b) This should only be updated if the existing explanations are either empty or the default value, right? Otherwise you may overwrite creator-supplied data without them noticing.

(c) Can you find a way to do this without $broadcast -- perhaps by moving more of the logic to a service? We are trying to remove broadcasts as part of the migration to Angular.

Reply:
(a) I had to create the HTML in the js file itself as it represents an RTE content, for which even if we represent it correctly in the HTML file, we need the 'p' tags to be saved explicitly in the variable also. For that, I used `document.createElement` and other in-built JS functions to achieve the same result (so that it is safer). Is that fine?

(b) Done

(c) Used SkillCreationService and a variable that iterates between 1 and -1 so that the directive can watch that and update as required. Also, it sets it to 0 to achieve (b).

Reply:
Yep, what you've done in (a) is fine. Thanks!